agentendpoint/inventory.go

```diff
@@ -0,0 +1,260 @@
+			})
+		}
+	}
+	// Map Deb packages to Apt packages.
+	if packages.Deb != nil {
+		for _, pkg := range packages.Deb {
+			softwarePackages = append(softwarePackages, &agentendpointpb.Inventory_SoftwarePackage{
+				Details: formatAptPackage(pkg),
+			})
+		}
+	}
+	// Map Rpm packages to Yum or Zypper packages depending on the OS.
+	if packages.Rpm != nil {
+		if shortName == "sles" {
+			for _, pkg := range packages.Rpm {
+				softwarePackages = append(softwarePackages, &agentendpointpb.Inventory_SoftwarePackage{
+					Details: formatZypperPackage(pkg),
+				})
+			}
+		} else {
+			for _, pkg := range packages.Rpm {
+				softwarePackages = append(softwarePackages, &agentendpointpb.Inventory_SoftwarePackage{
+					Details: formatYumPackage(pkg),
+				})
+			}
+		}
+	}
+	// Ignore Pip and Gem packages.
+
+	return softwarePackages
+}
+
+func formatAptPackage(pkg packages.PkgInfo) *agentendpointpb.Inventory_SoftwarePackage_AptPackage {
+	return &agentendpointpb.Inventory_SoftwarePackage_AptPackage{
+		AptPackage: &agentendpointpb.Inventory_VersionedPackage{
+			Name:         pkg.Name,
+			Architecture: pkg.Arch,
+			Version:      pkg.Version,
+		}}
+}
+
+func formatGooGetPackage(pkg packages.PkgInfo) *agentendpointpb.Inventory_SoftwarePackage_GoogetPackage {
+	return &agentendpointpb.Inventory_SoftwarePackage_GoogetPackage{
+		GoogetPackage: &agentendpointpb.Inventory_VersionedPackage{
+			Name:         pkg.Name,
+			Architecture: pkg.Arch,
+			Version:      pkg.Version,
+		}}
+}
+
+func formatYumPackage(pkg packages.PkgInfo) *agentendpointpb.Inventory_SoftwarePackage_YumPackage {
+	return &agentendpointpb.Inventory_SoftwarePackage_YumPackage{
+		YumPackage: &agentendpointpb.Inventory_VersionedPackage{
+			Name:         pkg.Name,
+			Architecture: pkg.Arch,
+			Version:      pkg.Version}}
+}
+
+func formatZypperPackage(pkg packages.PkgInfo) *agentendpointpb.Inventory_SoftwarePackage_ZypperPackage {
+	return &agentendpointpb.Inventory_SoftwarePackage_ZypperPackage{
+		ZypperPackage: &agentendpointpb.Inventory_VersionedPackage{
+			Name:         pkg.Name,
+			Architecture: pkg.Arch,
+			Version:      pkg.Version}}
+}
+
+func formatZypperPatch(pkg packages.ZypperPatch) *agentendpointpb.Inventory_SoftwarePackage_ZypperPatch {
+	return &agentendpointpb.Inventory_SoftwarePackage_ZypperPatch{
+		ZypperPatch: &agentendpointpb.Inventory_ZypperPatch{
+			Name:     pkg.Name,
+			Category: pkg.Category,
+			Severity: pkg.Severity,
+			Summary:  pkg.Summary,
+		}}
+}
+
+func formatWUAPackage(pkg packages.WUAPackage) *agentendpointpb.Inventory_SoftwarePackage_WuaPackage {
+	var categories []*agentendpointpb.Inventory_WindowsUpdatePackage_WindowsUpdateCategory
+	for idx, category := range pkg.Categories {
+		categories = append(categories, &agentendpointpb.Inventory_WindowsUpdatePackage_WindowsUpdateCategory{
+			Id:   pkg.CategoryIDs[idx],
+			Name: category,
+		})
+	}
+
+	// TODO: Populate supportUrls with MoreInfoUrls.
+	supportUrls := []string{}
```

Comment:
@adjackura Added a TODO here and setting support URLs to be empty list in the meantime